src/linux/linux-host-session.vala

```diff
@@ -419,7 +408,47 @@ namespace Frida {
 			return result;
 		}
 
-		public async uint spawn (string package_name, string? activity_name) throws Error {
+		public async uint spawn (string program, HostSpawnOptions options) throws Error {
+
+			string package_name = program;
+
+			var aux_options = options.load_aux ();
+
+			string  method = "activity";
+			string? entry_point = null;
+			if (aux_options.contains ("activity")) {
+
+				string? activity_name = null;
+
+				if (!aux_options.lookup ("activity", "s", out activity_name))
+					throw new Error.INVALID_ARGUMENT ("The 'activity' option must be a string");
+
+				if (activity_name[0] == '.')
+					activity_name = package_name + activity_name;
+
+				entry_point = activity_name;
+			}
+			if (aux_options.contains ("receiver")) {

```

Comment:
Perhaps we should enforce mutual exclusion here so we throw an error if both `activity` and `receiver` are used? (And any invalid combination of the three.)

Reply:
ok